feat: Priest in Peril quest (377)

MrAshieB opened the pull request.

MrAshieB committed:
feat: add some priest in peril config

MrAshieB committed:
feat: add some priest in peril config

MrAshieB committed:
chore: add more configs for priest in peril

MrAshieB committed:
chore: add more configs for priest in peril

MrAshieB committed:
Merge branch 'priestperil' of https://github.com/MrAshieB/Content into priestperil

MrAshieB committed:
feat: Priest in Peril quest

Missing dialogue for Wolfbane reclamation currently.

MrAshieB committed:
fix: pip-correct authentic Drezel dialogue with source

MrAshieB committed:
fix: pip-authentic messages instead of mesboxes for monuments

trufitus commented on scripts/areas/area_mausoleum/scripts/drezel.rs2:
```diff
@@ -0,0 +1,101 @@
+    ~chatplayer("<p,quiz>Why, what's happened?");
+    ~chatnpc("<p,sad>From what I can tell, after you killed the guard dog who protected the entrance to the monuments, those Zamorakians force the door into the main chamber");
+    ~chatnpc("<p,sad>and have used some kind of evil potion upon the well which leads to the source of the river Salve. As they have done this at the very mouth of the river");
+    ~chatnpc("<p,sad>it will spread along the entire reiver, disrupting the blessing placed upon it and allowing the evil creatures of Morytania to invade at their leisure.");
```
typo? Dialogue seems correct on osrs 2013

MrAshieB replied:
Thanks - resolved! :)

trufitus commented on scripts/areas/area_mausoleum/scripts/drezel.rs2:
```diff
@@ -0,0 +1,101 @@
+    ~chatnpc("<p,neutral>find some kind of way of removing or counteracting the evil magic that has been put into the river source at the well, so that the river will flow pure once again.");
+    ~chatplayer("<p,quiz>Couldn't you bless the river to purify it? Like you did with the water I took from the well?");
+    ~chatnpc("<p,sad>No, that would not work, the power I have from Saradomin is not great enough to cleanse an entire river of this foul Zamorakian pollutant.");
+    ~chatnpc("<p,neutral>I have only one idea how we could possibly cleanse the river?");
```
Wrong punctuation at the end?

MrAshieB replied:
Thanks - resolved! :)

trufitus commented on scripts/areas/area_mausoleum/scripts/drezel.rs2:
```diff
@@ -0,0 +1,101 @@
+    }
+}
+
+[label,priestperil_drezel]
```
You could break this label up by the different quest stages and do the if/else chain in the opnpc1

MrAshieB replied:
Thanks - resolved! :)

trufitus commented on scripts/areas/area_mausoleum/scripts/drezel.rs2:
```diff
@@ -0,0 +1,101 @@
+}
+
+[label,drezel_give_blankrunes]
+if(inv_total(inv, blankrune) > 0 & %priestperil ! ^priestperil_end_bring_essence) {
```
suggestion: invert the if condition and return early to avoid indenting a large code block

trufitus commented on scripts/areas/area_mausoleum/scripts/drezel.rs2:
```diff
@@ -0,0 +1,101 @@
+~chatplayer("<p,neutral>Okay, I will keep it equipped whenever I fight werewolves.");   
+
+[label,reclaim_wolfbane_dagger]
+//TODO: Dialogue for reclaiming Wolfbane dagger. Unable to find any sources yet.
```
Can always reference the wiki transcripts and leave a comment that you're not sure it's authentic dialogue
https://runescape.wiki/w/Transcript:Priest_in_Peril#Post-Quest_Dialogue

MrAshieB replied:
Little unsure on this one...
Added some sources I used, though.

MrAshieB replied:
No source on dialogue for reclaiming the dagger, that I can find. Unfortunately I don't have membership anymore to try and find out what OSRS does with the reclamation. RS3 has an opcode on the NPC to reclaim via an interface, with no dialogue so can't use that...

Indio3 replied:
from osrs, first has no mesanim, rest are neutral

https://github.com/user-attachments/assets/b7a1f223-cd70-46bd-9737-b5d8747ae6c5

```
"I've lost my wolfbane dagger."
"Yes, I know! Luckily for you it washed up on the banks of the Salve earlier! Here, take it again, but please try and be more careful this time."
"It's a family heirloom after all!"
"Thanks for that!"
```

MrAshieB replied:
Appreciate you fetching that for me 😄 ❤️

trufitus commented on scripts/areas/area_mausoleum/scripts/drezel.rs2:
```diff
@@ -0,0 +1,101 @@
+~chatnpc("<p,happy>it has the power to prevent werewolves from adopting their wolf form in combat as long as you have it equipped.");
+queue(priestperil_complete, 0);
+
+[label,drezel_dialogue]
```
This dialogue only happens once and then the varp is set to 61, unlocking access to the holy barrier. Before that if you try to cross he yells at you to talk to him.

MrAshieB replied:
Thanks for highlighting this to me!

Is this true in 2004-2006 too? I understand it is in OSRS, at least.
I can't find any source for this; do you have one I can review, perhaps?

thesuddensilent replied:
Have you checked the rs3 version? How does it work there? Some things may still be unchanged there compared to the osrs side.

MrAshieB replied:
Hmm, I guess I could set varp to 61 after the 'So can I pass through that barrier now?' dialogue... Which unlocks the Holy Barrier. Which would then allow the 'Greetings again adventurer. How go your travels in Moryania? Is it as evil as I have heard?' dialogue, from https://runescape.wiki/w/Transcript:Priest_in_Peril#Post-Quest_Dialogue

Whether that was authentic for 2004-2006, I'm not sure!

Indio3 replied:
60 is complete in 530 https://github.com/Joshua-F/cs2-scripts/blob/rs2-530/scripts/%5Bproc,script5%5D.cs2#L554, not sure if osrs still matches that (does RSprox show it being set to 61 when you get the completion interface? I'm guessing it probably sets to 60 when you hand in last ess and queues the completion at that point)

MrAshieB replied:
Ahh, so when I recorded this in RSprox, the quest completed at 60, with the interface too. But then you talk to Drezel and ask 'So can I pass through that barrier now?', and it upped it to 61 with Drezel giving a 'blessing'.


```
if_settext | com=chat_left:name (231:4), text="Drezel"
if_settext | com=chat_left:text (231:6), text="Right, I think that's everything. Just the blessing itselfnow. Hold still a moment."

varp_small | id=priestperil (302), old=60, new=61
message_game | type=mesbox, message="Drezel blesses you, allowing you to pass through the barrier into Morytania."
spotanim | id=heal_casting (84), height=120
sequence | id=human_castentangle (710)
```

AmVoidGuy replied:
Yeah, the quest completes at 60. Osrs checks 60 for getting museum kudos for instance, 61 is used for checking access to Morytania for shortcuts / teleports. Quest completes at 60 but you can't enter Morytania in anyway until talking to him again and unlocking the barrier. The barrier exists back in 377 so I'd assume it still functions the same until a source can prove otherwise

MrAshieB replied:
Sounds good to me - I've implemented this now... Hopefully that's a bit better 😄 
Thanks!

MrAshieB committed:
fix: Typos and reorg Drezel conversations

MrAshieB committed:
feat: add wolfbane dagger reclamation

MrAshieB committed:
fix: add more dialogue before being able to use holy barrier

Also includes starting dialogue for Nature Spirit, to be followed up with a Nature Spirit quest PR.

MrAshieB committed:
fix: add message when passing through holy barrier
